gdk4/src/texture.rs

```diff
@@ -18,3 +21,47 @@ impl<O: IsA<Texture>> TextureExtManual for O {
         }
     }
 }
+
+impl Texture {
+    pub fn for_pixbuf_future(
+        pixbuf: &gdk_pixbuf::Pixbuf,
+    ) -> Pin<Box<dyn future::Future<Output = Result<Texture, glib::Error>> + 'static>> {
+        Box::pin(gio::GioFuture::new(
+            &(),
+            glib::clone!(@weak pixbuf => move |_, _, send| {
+                let res = Texture::for_pixbuf(pixbuf);
+                send.resolve(res);
+            }),
+        ))
+    }
+
+    pub fn from_bytes_future(
+        bytes: &glib::Bytes,
+    ) -> Pin<Box<dyn future::Future<Output = Result<Texture, glib::Error>> + 'static>> {
```

Comment:
As those are not in a trait, why make use of Pin<Box<dyn future... instead of just marking the function as `pub async fn`?